core-java-modules/core-java-graphql/src/main/java/com/baeldung/graphql/data/BookRepository.java

```diff
@@ -0,0 +1,23 @@
+package com.baeldung.graphql.data;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class BookRepository {
+
+    private static List<Book> books = null;
+
+    public BookRepository() {
+        if (books == null) {
```

Comment:
Oooh - we've got a constructor of an instance object that's actually doing initialization of a static member. That's a bit confusing. Let's do this in the initializer of the field itself:

```java
private static List<Book> books = Stream.of(
    new Book(...), 
    new Book(...))
  .collect(toList()));
```